core/src/main/scala/org/genivi/sota/core/DevicesResource.scala

```diff
@@ -107,8 +109,36 @@ class DevicesResource(db: Database, client: ConnectivityClient,
     }
   }
 
+  def getDeviceWithStatus(ns: Namespace, uuid: Uuid): Route = {
+    parameters(('status.?(false))) { (includeStatus: Boolean) =>
+      val devices = deviceRegistry.fetchDevice(uuid).flatMap { device =>
+        if (device.namespace == ns) {
+          Future.successful(Seq(device))
+        } else {
+          Future.failed(MissingEntity(classOf[DeviceSearchResult]))
```

Comment:
Is it really a MissingEntity, if the device exists but is of the wrong namespace?